File: 189-rotate-array/rotate-array.ts
/**
 Do not return anything, modify nums in-place instead.
 */
function rotate(nums: number[], k: number): void {
    const arr=new Array(nums.length);
    let n:number=nums.length;
    let i:number;
    for(i=0;i<nums.length;i++){
        let index:number=(i+k)%n;
        arr[index]=nums[i];
    }
     for (let i = 0; i < n; i++) {
        nums[i] = arr[i];
    }

};

File: 189-rotate-array/rotate-array.py
class Solution(object):
    def revers(self,l,r,nums):
        while(l<r):
            nums[l],nums[r]=nums[r],nums[l]
            l+=1;
            r-=1;
    def rotate(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        n=len(nums);
        k=k%n;
        self.revers(n-k,n-1,nums);
        self.revers(0,n-k-1,nums);
        self.revers(0,n-1,nums);
        